internal/util/util.go

```diff
@@ -135,54 +136,51 @@ func GetVolumeSnapshotClassForStorageClass(provisioner string, snapshotClient sn
 }
 
 // GetVolumeSnapshotContentForVolumeSnapshot returns the volumesnapshotcontent object associated with the volumesnapshot
-func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1beta1api.VolumeSnapshot, snapshotClient snapshotter.SnapshotV1beta1Interface, log logrus.FieldLogger, wait bool) (*snapshotv1beta1api.VolumeSnapshotContent, error) {
+func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1beta1api.VolumeSnapshot, snapshotClient snapshotter.SnapshotV1beta1Interface, log logrus.FieldLogger, shouldWait bool) (*snapshotv1beta1api.VolumeSnapshotContent, error) {
```

Comment:
I think I would handle the `shouldWait == false` case on its own at the top of this function (outside the loop), for a few reasons:
- makes the polling simpler since you don't have to deal with that case
- I don't see a need to query the API again for the `VolumeSnapshot` - we can just use the version that's been passed into the plugin
- if it doesn't have a `.Status.BoundVolumeSnapshotContentName`, or that VolumeSnapshotContent doesn't have a `snapshotHandle` that shouldn't be an error since we're not waiting for them to provision, **which is different** than for Velero-created VolumeSnapshots

Something like the following:

```go
if !shouldWait {
    if volSnap.Status == nil || volSnap.Status.BoundVolumeSnapshotContentName == nil {
        // no bound VSC: it's not an error, but there's nothing additional to restore
        return nil, nil
    }

    vsc, err := snapshotClient.VolumeSnapshotContents().Get(*volSnap.Status.BoundVolumeSnapshotContentName, metav1.GetOptions{})
    if err != nil {
        return nil, errors.Wrap(err, "error getting volume snapshot content from API")
    }

    return vsc, nil
}
```

Note, you'll also have to handle nil `VolumeSnapshotContent` and nil `VolumeSnapshotContent.Status.SnapshotHandle` in the calling function if you use this approach.

Then, you can remove having to worry about that case from the polling below.

Reply:
I used your suggestion. But, IMO, I don't think it made the polling logic any simpler.
The only part in the polling that was waiting aware was a log message.